semgrep-core/src/analyzing/AST_to_IL.ml

```diff
@@ -579,11 +579,19 @@ and expr_aux env ?(void = false) eorig =
                 ss_for_e3 @ [ mk_s (Instr (mk_i (Assign (lval, e3)) e3orig)) ]
               )));
       lvalexp
+  | G.Await (tok, e) ->
+      let lval = fresh_lval env (G.fake "await") in
```

Comment:
Shouldn't we use `tok` instead of `G.fake`? Just want to make sure this wouldn't cause a crash if we tried to convert to a range later.

Reply:
True, good catch.

Reply:
Otherwise, you could perhaps use `Parse_info.fake_info` to get a "safe fake token". And we should make `AST_generic.fake` an alias of that one.